pkg/core/native/management.go

```diff
@@ -0,0 +1,72 @@
+package native
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+)
+
+// Management is contract-managing native contract.
+type Management struct {
+	interop.ContractMD
+
+	lock    sync.RWMutex
+	isValid bool
+	nextID  int32
+}
+
+const (
+	managementName        = "Neo Contract Management"
+	prefixContract        = 8
+	prefixNextAvailableId = 15
+)
+
+// newManagement creates new Management native contract.
+func newManagement() *Management {
+	var m = &Management{ContractMD: *interop.NewContractMD(managementName)}
+
+	return m
+}
+
+// Metadata implements Contract interface.
+func (m *Management) Metadata() *interop.ContractMD {
+	return &m.ContractMD
+}
+
+// OnPersist implements Contract interface.
+func (m *Management) OnPersist(ic *interop.Context) error {
+	if ic.Block.Index != 0 { // We're only deploying at 0 at the moment.
```

Comment:
If I'm not mistaken, this `0` is for C#'s `ActiveBlockIndex`. There's one thing I doubt about: [this line](https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/ApplicationEngine.Contract.cs#L98) in C# code. If `ActiveBlockIndex` for any native contract will be changed in C#, we may easily forget to add this check to `native.Call`.

Reply:
For already existing contracts it's not really possible, that's why I've dropped this logic. And for new ones (added post-3.0) it shouldn't be a problem because they won't be deployed before `ActiveBlockIndex` (thus `GetContract` will just fail for them leading to call failure).